magento/pwa-studio
# [spike] Add focus trapping with react-aria

## Description

Use Adobe's `react-aria` library to implement focus trapping in dialogs. The issue calls for the `useDialog` hook, but this spike currently uses the `FocusScope` component instead.

#### New libraries

- `react-aria`

Module | Size (gzip) | Size (parse)
:---|---:|---:
`SSRProvider` | `0.35 kB` | `0.62 kB`
`FocusScope` | `5.38 kB` | `19.05 kB`

## Related Issue
Closes PWA-619.

## Acceptance

### Verification Stakeholders

- @schensley 
- @supernova-at 

### Specification

### Verification Steps


1. Go to the checkout page.
2. Hit `Tab` a dozen times to focus different elements. Focus should cycle between children of the dialog.

Only valid focusable elements should receive focus.

## Screenshots / Screen Captures (if appropriate)

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
